sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -136,6 +137,7 @@ public void Dispose()
         /// Marks the scope as failed.
         /// </summary>
         /// <param name="exception">The exception to associate with the failed scope.</param>
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "The Exception being passed into this method has the commonly used properties being preserved with DynamicallyAccessedMemberTypes on the inner method MarkFailed.")]
```

Comment:
There's a warning coming from this method since the internal method MarkFailed preserves the public properties on the Exception type:
```
C:\...\DiagnosticScope.cs(142): Trim analysis warning IL2026: Azure.Core.Pipeline.DiagnosticScope.Failed(Exception): Using member 'System.Exception.TargetSite.get' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Metadata for the method might be incomplete or removed. 
```

Reply:
I don't believe System.Exception.TargetSite.get is needed by ApplicationInsights. I'm not sure if there is a better way to resolve this warning than suppressing it.

Reply:
Agreed. Honestly even if AppInsights would use that property, there's nothing we co do about it - that property is simply not going to work as on CoreCLR. AppInsights would have to adapt to that reality.

This one we do need to suppress - with a good description of what's going on, and why it's OK to suppress.